frontend/static/js/pong/Light.js

```diff
@@ -0,0 +1,42 @@
+import NotImplentedError from "../exceptions/NotImplementedException.js";
+import * as THREE from 'three';
+
+export default class Light {
+	constructor(Light, {color, intensity, position, circleRadius, angleRad, speed, target}) {
+		this.position = position;
+		this.angleRad = angleRad || null;
+		this.speed = speed || null;
+		this.color = color;
+		this.intensity = intensity;
+		this.lightCircleRadius = circleRadius || null;
+		this.light = this.createLight(Light);
+		this.target = target || null;
+	}
+
+	createLight(Light) {
+		const light = new Light(this.color, this.intensity);
+		light.position.set(this.position.x, this.position.y, this.position.z);
+		light.castShadow = true;
+		if (this.target) {
+			light.target = this.target;
+		}
+		return light;
+	}
+
+	followObject(target) {
+		if (!(this.light instanceof THREE.SpotLight)) {
+			throw new TypeError('Light must be a SpotLight to follow an object.');
+		}
+		this.light.target = target;
+		this.light.target.updateMatrixWorld();
+	}
+
+	moveLightCircular() {
+		if (this.angleRad === null || this.speed === null) {
+			throw new NotImplentedError('angleRad and speed must be set to move light circularly.');
```

Comment:
a small thing but NotImplementedException is a quite special exception in specifically usually indicates a future feature for a feature! Like for example now that we inherit the ARequestService we are lazy and use that to block some function calls through services that dont handle a specific request / we wont expose a specific api call with specific service in this project -> check out GET from FriendService!

More appropriate might be something like ArgumentException (assuming we wont really ever get here and wont show the error to the user) - tells the coder that you are trying to use this function with faulty arguments.

Reply:
I could write our own error for that (JS doesn't come with one out of the box). I would do it with the more centralised error handling in another PR